__tests__/pages/exercises/[lessonSlug].test.js

```diff
@@ -1,12 +1,122 @@
 import React from 'react'
-import { render, waitFor } from '@testing-library/react'
+import {
+  render,
+  waitFor,
+  screen,
+  waitForElementToBeRemoved
+} from '@testing-library/react'
 import '@testing-library/jest-dom'
 import Exercises from '../../../pages/exercises/[lessonSlug]'
+import { useRouter } from 'next/router'
+import { MockedProvider } from '@apollo/client/testing'
+import dummyLessonData from '../../../__dummy__/lessonData'
+import dummySessionData from '../../../__dummy__/sessionData'
+import dummyAlertData from '../../../__dummy__/alertData'
+import GET_APP from '../../../graphql/queries/getApp'
+
+const session = {
+  ...dummySessionData,
+  submissions: [],
+  lessonStatus: []
+}
 
 describe('Exercises page', () => {
+  const { query } = useRouter()
+  query['lessonSlug'] = 'js0'
   test('Should render correctly', async () => {
-    const { getByRole } = render(<Exercises />)
+    const mocks = [
+      {
+        request: { query: GET_APP },
+        result: {
+          data: {
+            session,
+            lessons: dummyLessonData,
+            alerts: dummyAlertData
+          }
+        }
+      }
+    ]
+
+    const { getByRole } = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Exercises />
+      </MockedProvider>
+    )
+
+    await waitFor(() =>
+      getByRole('heading', { name: /Foundations of JavaScript/i })
```

Comment:
_NIT_

I'm not really sure if this is important because it's okay for tests to have repeated code, inconsistency or a literally a mess; but `screen` is already imported. It could be used in place of `{ getByRole } = render()` -> `screen.getByRole`.

Reply:
Good idea! Screen checks the whole screen. My only concern is I'm not sure how this will affect test speed. It's probably fine though.